src/UI/Actions.hs

```diff
@@ -860,7 +865,14 @@ abort = Action ["cancel"] (reset (Proxy :: Proxy v) (Proxy :: Proxy a))
 focus :: forall a v. (HasViewName v, HasName a, Focusable v a) => Action v a AppState
 focus = Action
   ["switch mode to " <> T.pack (show (name (Proxy :: Proxy a)))]
-  (switchFocus (Proxy :: Proxy v) (Proxy :: Proxy a))
+  (\s -> do
+    liftIO . view (asConfig . confLogSink) s . T.pack $
+      "switchFocus "
```

Comment:
When I'm going a big concat I tend to `mconcat` a list. Maybe you might like how that looks?

Reply:
all reduces the same program :)  it's just an aesthetic thing.

If I want to go wild I'd sooner switch to something using `Data.Text.Builder` so that the logging framework can stream it straight into the file (well, maybe it already fuses!  I haven't read the core yet...)